pkg/api/staking_test.go

```diff
@@ -0,0 +1,192 @@
+			jsonhttptest.WithExpectedJSONResponse(&jsonhttp.StatusResponse{Code: http.StatusBadRequest, Message: "invalid staking amount"}))
+	})
+
+	t.Run("gas limit header", func(t *testing.T) {
+		t.Parallel()
+
+		var gasLimit uint64
+		contract := stakingContractMock.New(
+			stakingContractMock.WithDepositStake(func(ctx context.Context, stakedAmount big.Int, overlay swarm.Address) error {
+				gasLimit = sctx.GetGasLimit(ctx)
+				return nil
+			}),
+		)
+		ts, _, _, _ := newTestServer(t, testServerOptions{
+			DebugAPI:        true,
+			StakingContract: contract,
+		})
+
+		jsonhttptest.Request(t, ts, http.MethodPost, depositStake(addr.String(), minStake), http.StatusOK,
+			jsonhttptest.WithRequestHeader("Gas-Limit", "2000000"),
+		)
+
+		if gasLimit != 2000000 {
+			t.Fatalf("want 2000000, got %d", gasLimit)
+		}
+	})
+}
+
+func TestGetStake(t *testing.T) {
+	t.Parallel()
+
+	addr := swarm.MustParseHexAddress("f30c0aa7e9e2a0ef4c9b1b750ebfeaeb7c7c24da700bb089da19a46e3677824b")
+
+	getStake := func(address string) string {
+		return fmt.Sprintf("/stake/%s", address)
+	}
+
+	t.Run("ok", func(t *testing.T) {
+		t.Parallel()
+
+		contract := stakingContractMock.New(
+			stakingContractMock.WithGetStake(func(ctx context.Context, overlay swarm.Address) (big.Int, error) {
+				return *big.NewInt(1), nil
+			}),
+		)
+		ts, _, _, _ := newTestServer(t, testServerOptions{DebugAPI: true, StakingContract: contract})
+		jsonhttptest.Request(t, ts, http.MethodGet, getStake(addr.String()), http.StatusOK,
+			jsonhttptest.WithExpectedJSONResponse(&api.GetStakeResponse{StakedAmount: big.NewInt(1)}))
+	})
+
+	t.Run("with invalid address", func(t *testing.T) {
+		t.Parallel()
+
+		ts, _, _, _ := newTestServer(t, testServerOptions{DebugAPI: true})
+		jsonhttptest.Request(t, ts, http.MethodGet, getStake("invalid address"), http.StatusBadRequest,
+			jsonhttptest.WithExpectedJSONResponse(&jsonhttp.StatusResponse{Code: http.StatusBadRequest, Message: "invalid address"}))
+	})
+
+	t.Run("with error", func(t *testing.T) {
+		t.Parallel()
+
+		contractWithError := stakingContractMock.New(
+			stakingContractMock.WithGetStake(func(ctx context.Context, overlay swarm.Address) (big.Int, error) {
+				return *big.NewInt(0), stakingcontract.ErrGetStakeFailed
+			}),
+		)
+		ts, _, _, _ := newTestServer(t, testServerOptions{DebugAPI: true, StakingContract: contractWithError})
+		jsonhttptest.Request(t, ts, http.MethodGet, getStake(addr.String()), http.StatusInternalServerError,
+			jsonhttptest.WithExpectedJSONResponse(&jsonhttp.StatusResponse{Code: http.StatusInternalServerError, Message: "get staked amount failed"}))
+	})
+	t.Run("gas limit header", func(t *testing.T) {
+		t.Parallel()
+
+		var gasLimit uint64
+		contract := stakingContractMock.New(
+			stakingContractMock.WithGetStake(func(ctx context.Context, overlay swarm.Address) (big.Int, error) {
+				gasLimit = sctx.GetGasLimit(ctx)
+				return *big.NewInt(1), nil
```

Comment:
could `gasLimit` be asserted here in mock stake method (right after `sctx.GetGasLimit(ctx)`) ?